src/main_process.py

```diff
@@ -9,12 +9,12 @@
 
 class MainProcess():
 
-    def __init__(self, email, password, demo):
+    def __init__(self, email=None, password=None, demo=None, completed_groups=[]):
```

Comment:
Don't set variable defaults with iterables. Reccomend setting completed_groups to None then changing to [] in the code if you truly want an empty array as its default. Python does a weird thing where if you have multiple processes of the same thing happening and you initiated with (or set the default to) an iterable, every action will be applied to every instance of that variable, which can lead to some frustrating behavior. Google it if you want to know more, I don't know exactly how or why it does that.

Reply:
Good suggestion! I just made that change